test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -916,41 +951,96 @@ func verifyDisksAttachedOnSameNode(t *testing.T, testState *kubevirtTestState) {
+					testState.appCtx.App.Key))
+			return
+		}
+		// for multiDisk VM we have to restart the VM since the volumes may be attached on different nodes and
+		// live-migration does not move the attachments
+		kvCli := kubevirt.Instance().GetKubevirtClient()
+		log.InfoD("Restarting a multi-disk VM %s/%s to make it bind-mounted", testState.vmPod.Namespace, testState.vmiName)
+		err = kvCli.VirtualMachine(testState.vmPod.Namespace).Restart(testState.vmiName, &kubevirtv1.RestartOptions{})
+		log.FailOnError(t, err, "Failed to restart VM %s/%s", testState.vmPod.Namespace, testState.vmiName)
+		waitForVMIReady(t, testState, 15*time.Minute)
+		// refresh state after VM restart
+		gatherInitialVMIInfo(t, testState)
+		success, err = allVMDisksBindMounted(testState)
+		log.FailOnError(t, err, "Failed to verify bind mount for %s", testState.appCtx.App.Key)
+		Dash.VerifyFatal(t, success, true, fmt.Sprintf(
+			"usePopulator is true but multi-disk vm disks are not bind-mounted even after restart for %s",
+			testState.appCtx.App.Key))
+		return
+	}
 
-		if initialCheck {
-			// This is the first time after VM was provisioned. It should be bind-mounted immediately. No need to retry.
-			Dash.VerifyFatal(t, isBindMounted(), true, fmt.Sprintf("Initial check for %s", vmDisk))
-			continue
+	// Wait for PX to perform additional live migration/s to return the VM to a bind-mounted state.
+	require.Eventuallyf(t, func() bool {
+		success, err := allVMDisksBindMounted(testState)
+		if err != nil {
+			log.Warn("Failed to verify bind mount for %s: %v", testState.appCtx.App.Key, err)
+			return false
 		}
+		if !success {
+			log.Warn("Waiting for VM disks to switch to bind-mount for %s", testState.appCtx.App.Key)
+			return false
+		}
+		return true
+	}, 10*time.Minute, 30*time.Second, fmt.Sprintf(
+		"VM disk/s did not switch to a bind-mount for %s", testState.appCtx.App.Key))
+}
+
+func allVMDisksBindMounted(testState *kubevirtTestState) (bool, error) {
+	var err error
+
+	// refresh VM pod
+	testState.vmPod, err = getVMPod(testState.appCtx, testState.vmDisks[0].volume)
+	if err != nil {
+		// this is expected while the live migration is running since there will be 2 VM pods
+		log.InfoD("Could not get VM pod for %s for context %s: %v", testState.vmDisks[0], testState.appCtx.App.Key, err)
+		return false, err
+	}
 
-		// Wait for PX to perform additional live migration/s to return the VM to a bind-mounted state.
-		require.Eventuallyf(t, isBindMounted, 10*time.Minute, 30*time.Second, "%s did not switch to a bind-mount", vmDisk)
+	for _, vmDisk := range testState.vmDisks {
+		log.InfoD("Checking bind mount for %s", vmDisk)
+		mountType, err := getVMDiskMountType(testState.vmPod, vmDisk)
+		if err != nil {
+			log.Warn("Failed to get mount type of %s: %v", vmDisk, err)
+			return false, err
+		}
+		if mountType != mountTypeBind {
+			log.Warn("%s is not bind-mounted yet (mountType=%q)", vmDisk, mountType)
```

Comment:
Printing the vmName or vmID here would make debugging easier.

Reply:
> Additionally printing the vmName or vmID here would make debugging easier.

added the namespace (app context) to the error message